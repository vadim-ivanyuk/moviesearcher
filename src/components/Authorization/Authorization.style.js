import styled from "styled-components";
import { theme } from "../../theme";

export const AuthButton = styled.button`
  cursor: pointer;
  color: ${theme.colorOrange};
  border: 1px solid ${theme.colorOrange};
  font-weight: 400;
  font-size: 14px;
  background: transparent;
  padding: 0.3rem 0.75rem;
  margin-right: ${(props) => props.right && "5px"};
  transition: all 0.3s;

  &:hover {
    background: ${theme.colorOrange};
    color: ${theme.colorDark};
  }
`;

export const AuthButtonLink = styled.a`
  color: inherit;

  &:hover {
    color: inherit;
  }
`;
