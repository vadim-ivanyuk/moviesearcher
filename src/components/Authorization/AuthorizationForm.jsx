import React from "react";
import { Field, reduxForm } from "redux-form";
import { required, minLength, indentPassword } from "../../utils/validate";
import { CustomField } from "../../elements/CustomField";

function AuthorizationForm(props) {
  const { handleSubmit } = props;

  const onSubmit = () => {};

  return (
    <>
      <p className="authorization-form-title">Авторизация</p>
      <form onSubmit={handleSubmit(onSubmit)} className="authorization-form">
        <Field
          name="login"
          component={CustomField}
          label="Логин"
          validate={[required, minLength]}
        />
        <Field
          name="password"
          type="password"
          component={CustomField}
          label="Пароль"
          validate={[required, minLength]}
        />
        <Field
          name="repeatPassword"
          type="password"
          component={CustomField}
          label="Повторите пароль"
          validate={[required, minLength, indentPassword]}
        />
        <button type="submit" className="btn btn-success w-100 mt-2 mb-3">
          Отправить
        </button>
      </form>
    </>
  );
}

AuthorizationForm = reduxForm({
  form: "AuthorizationForm",
})(AuthorizationForm);

export { AuthorizationForm };
