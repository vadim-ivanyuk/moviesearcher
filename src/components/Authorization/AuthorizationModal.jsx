import React from "react";
import { Modal, ModalBody } from "reactstrap";
import PropTypes from "prop-types";
import { AuthorizationForm } from "./AuthorizationForm";

export const AuthorizationModal = (props) => {
  const { modal, toggle } = props;

  return (
    <Modal isOpen={modal} toggle={toggle} className="authorization-form-modal">
      <ModalBody>
        <AuthorizationForm />
      </ModalBody>
    </Modal>
  );
};

AuthorizationModal.defaultProps = {
  modal: false,
};

AuthorizationModal.propTypes = {
  modal: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
};
