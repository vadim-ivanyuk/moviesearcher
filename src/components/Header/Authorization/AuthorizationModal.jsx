import React from "react";
import { Modal, ModalBody } from "reactstrap";
import AuthorizationForm from "./Form/AuthorizationForm";

export const AuthorizationModal = (props) => {
  const { modal, toggle } = props;

  return (
    <Modal isOpen={modal} toggle={toggle} className="authorization-form-modal">
      <ModalBody>
        <AuthorizationForm />
      </ModalBody>
    </Modal>
  );
};
