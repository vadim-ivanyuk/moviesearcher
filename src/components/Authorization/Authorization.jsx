import React, { Fragment, useState } from "react";
import { AuthorizationModal } from "./AuthorizationModal";
import { AuthButton, AuthButtonLink } from "./Authorization.style";

export const Authorization = () => {
  const [modal, setModal] = useState(false);

  const toggle = () => setModal(!modal);

  return (
    <Fragment>
      <Fragment>
        <AuthButton right onClick={toggle}>
          Войти
        </AuthButton>
        <AuthorizationModal modal={modal} toggle={toggle} />
      </Fragment>
      <AuthButton>
        <AuthButtonLink
          href="https://www.themoviedb.org/signup"
          rel="nofollow"
          target="blank"
        >
          Регистрация
        </AuthButtonLink>
      </AuthButton>
    </Fragment>
  );
};
