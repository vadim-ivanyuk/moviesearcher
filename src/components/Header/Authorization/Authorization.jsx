import React, { useState } from "react";
import { AuthorizationModal } from "./AuthorizationModal";

export const Authorization = () => {
  const [modal, setModal] = useState(false);

  const toggle = () => setModal(!modal);

  return (
    <>
      <div className="authorization">
        <button onClick={toggle} className="btn btn-outline-warning mr-2">
          Войти
        </button>
        <AuthorizationModal modal={modal} toggle={toggle} />
      </div>
      <button className="btn btn-outline-warning">
        <a
          href="https://www.themoviedb.org/signup"
          rel="nofollow"
          target="blank"
        >
          Регистрация
        </a>
      </button>
    </>
  );
};
