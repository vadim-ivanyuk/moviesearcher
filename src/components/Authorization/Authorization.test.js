import React from "react";
import { render } from "react-dom";
import { configure, shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { Authorization } from "./Authorization";

configure({ adapter: new Adapter() });

describe("Authorization", () => {
  it("snap", () => {
    const authorization = shallow(<Authorization />);

    expect(shallowToJson(authorization)).toMatchSnapshot();
  });
});
