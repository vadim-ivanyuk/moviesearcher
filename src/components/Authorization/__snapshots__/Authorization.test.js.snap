// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization snap 1`] = `
<Fragment>
  <div
    className="authorization"
  >
    <button
      className="btn btn-outline-warning mr-2"
      onClick={[Function]}
    >
      Войти
    </button>
    <AuthorizationModal
      modal={false}
      toggle={[Function]}
    />
  </div>
  <button
    className="btn btn-outline-warning"
  >
    <a
      href="https://www.themoviedb.org/signup"
      rel="nofollow"
      target="blank"
    >
      Регистрация
    </a>
  </button>
</Fragment>
`;
